### PR TITLE
Use custom transport for gitlab communication

### LINKED ISSUE #11689: Gitlab is not using custom transport
https://github.com/openshift/origin/blob/master/pkg/auth/oauth/external/gitlab/gitlab.go#L97

Gitlab idp is using the default client for looking up user info, instead of using the configured transport

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #11689

### CODE REVIEW

**Comment by liggitt** [comment]
@enj PTAL

File: pkg/auth/oauth/external/gitlab/gitlab.go [hunk 1]
```diff
@@ -94,7 +94,11 @@ func (p *provider) GetUserIdentity(data *osincli.AccessData) (authapi.UserIdenti
 	req, _ := http.NewRequest("GET", p.userAPIURL, nil)
```
Comment (enj) [inline pkg/auth/oauth/external/gitlab/gitlab.go]: Is it safe to ignore this `err`?

Changes to client / transport LGTM.

Comment (liggitt) [inline pkg/auth/oauth/external/gitlab/gitlab.go]: yes, userAPIURL is formed in NewProvider from a known good url and path

**Comment by liggitt** [comment]
[merge]

File: pkg/auth/oauth/external/gitlab/gitlab.go [hunk 2]
```diff
@@ -94,7 +94,11 @@ func (p *provider) GetUserIdentity(data *osincli.AccessData) (authapi.UserIdenti
...
 	req.Header.Set("Authorization", fmt.Sprintf("bearer %s", data.AccessToken))
 
-	res, err := http.DefaultClient.Do(req)
+	client := http.DefaultClient
+	if p.transport != nil {
+		client := &http.Client{Transport: p.transport}
```
Comment (enj) [inline pkg/auth/oauth/external/gitlab/gitlab.go]: d'oh missed it - should be `=` not `:=`

**Pull request merged**